core/src/avm2/value.rs

```diff
@@ -858,25 +859,14 @@ impl<'gc> Value<'gc> {
 
     /// Coerce the value to an object, and report an error relating to object
     /// receivers being null or undefined otherwise.
-    ///
-    /// If the `name` parameter is provided, the error will indicate the object
-    /// property being interacted with.
-    pub fn coerce_to_receiver(
+    pub fn coerce_to_object_or_typeerror(
         &self,
         activation: &mut Activation<'_, 'gc>,
-        name: Option<&Multiname<'gc>>,
     ) -> Result<Object<'gc>, Error<'gc>> {
-        self.coerce_to_object(activation).map_err(|_| {
-            if let Some(name) = name {
-                format!(
-                    "Cannot access property {} of null or undefined",
-                    name.to_qualified_name(activation.context.gc_context)
```

Comment:
I think it's really important that we keep the ability to show field names in the error message. One approach would be to store extra data in `ErrorObjectData`, and only display it in the `Debug` impl.

Reply:
I readded it to every message; if we wanna split it to a separate private field of Error/TypeError, let's do it in another PR, okay?